packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -38,30 +40,103 @@ class _TextSelectionToolbar extends StatelessWidget {
   final VoidCallback handlePaste;
   final VoidCallback handleSelectAll;
 
+  @override
+  __TextSelectionToolbarState createState() => __TextSelectionToolbarState();
+}
+
+class __TextSelectionToolbarState extends State<_TextSelectionToolbar> {
+  // effective when items > 3
+  bool flipToMenu = false;
+
   @override
   Widget build(BuildContext context) {
     final List<Widget> items = <Widget>[];
-    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-
-    if (handleCut != null)
-      items.add(FlatButton(child: Text(localizations.cutButtonLabel), onPressed: handleCut));
-    if (handleCopy != null)
-      items.add(FlatButton(child: Text(localizations.copyButtonLabel), onPressed: handleCopy));
-    if (handlePaste != null)
-      items.add(FlatButton(child: Text(localizations.pasteButtonLabel), onPressed: handlePaste,));
-    if (handleSelectAll != null)
-      items.add(FlatButton(child: Text(localizations.selectAllButtonLabel), onPressed: handleSelectAll));
+    final MaterialLocalizations localizations =
+        MaterialLocalizations.of(context);
+
+    if (widget.handleCut != null)
+      items.add(FlatButton(
+          child: Text(localizations.cutButtonLabel),
+          onPressed: widget.handleCut));
+    if (widget.handleCopy != null)
+      items.add(FlatButton(
+          child: Text(localizations.copyButtonLabel),
+          onPressed: widget.handleCopy));
+    if (widget.handlePaste != null)
+      items.add(FlatButton(
+        child: Text(localizations.pasteButtonLabel),
+        onPressed: widget.handlePaste,
+      ));
+    if (widget.handleSelectAll != null)
+      items.add(FlatButton(
+          child: Text(localizations.selectAllButtonLabel),
+          onPressed: widget.handleSelectAll));
 
     // If there is no option available, build an empty widget.
     if (items.isEmpty) {
       return Container(width: 0.0, height: 0.0);
     }
 
+    if (items.length <= 3) {
+      return _buildSelectionToolbar(items);
+    } else {
+      final List<Widget> displayItems = <Widget>[
+        items[0],
+        items[1],
+        items[2],
+        IconButton(
+          icon: const Icon(Icons.more_vert),
+          onPressed: () {
+            setState(() {
+              flipToMenu = true;
+            });
+          },
+        ),
+      ];
+
+      if (flipToMenu) {
+        return _buildSelectionMenu(localizations.selectAllButtonLabel);
+      } else {
+        return _buildSelectionToolbar(displayItems);
+      }
+    }
+  }
+
+  Widget _buildSelectionMenu(String selectAllButtonLabel) {
+    return Material(
+      elevation: 1.0,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          FlatButton(
```

Comment:
This should be dynamic per comment above.

Reply:
Since it is a column, it would only accept a list of Widgets. How can I change that definition to List<dynamic>?

Reply:
Maybe _buildSelectionMenu could receive a list of overflowing buttons, then you add the back button to that list.  Something like this:

```dart
Widget _buildSelectionMenu(List<FlatButton> buttons) {

  ...

    children: <Widget>[
      ...buttons,
      IconButton(
        icon: const Icon(Icons.arrow_back),
        ...
      ),
    ],
    ...
```

Then in `build`, determine which buttons overflow and pass them in when you call `_buildSelectionMenuToolbar`.